server.go

```diff
@@ -287,21 +322,73 @@ func (s *Server) runHandler(handler HandlerFunc, deliveries <-chan amqp.Delivery
 
 		ctx := context.WithValue(context.Background(), CtxQueueName, queueName)
 
-		// TODO: Recover panic in handler. And use delivery.Nack() so amqp will retry.
-		handler(ctx, &rw, delivery)
-		delivery.Ack(false)
+		// Process the request in a separate go routine to not crash the server
+		// on panic and make use of the panic recovery.
+		go func(delivery amqp.Delivery) {
```

Comment:
Nice! This makes requests to the same handler not block.

Reply:
So I think this is what we should keep from this PR :)